### ISSUE TITLE
[TEST] create a unit test to ensure our deep copy functions work correctly

### ISSUE THREAD

**Issue by joshimoo** [longhorn/longhorn#2634]
**What's the test to develop? Please describe.**
As I had to fix some deep copy issues before and it's easy to miss, we should ensure that all deep copy functions for our resources work correctly.

This should be able to be accomplished via reflection by walking each resource type and populating the resource with random data. Then deep copy -> deep equals + isDifferentReference check for each reference type of the original / copy.

I am guessing this is between a 3 and 5 in effort, since one doesn't usually work with reflection everyday and one might want to check if there are similar issues in the other longhorn components.

**Additional context**
Previously fixed deep copy issues: #1950 #2631

**Comment by yasker** [comment]
It's a minor effort that likely worth it. We only need to cover the longhorn manager since no other components deal with Kubernetes so doesn't need a deep copy, and `reflect` is not that hard. Kubernetes might already have a way to test it.

My estimate is it probably will take less than 3 days.

**Comment by joshimoo** [comment]
@yasker I agree should be do able in 3 days, I left some room for error though :)

**Comment by joshimoo** [comment]
This might be superseded by #2626 if we can update the code generation setup to auto gen deep copy functions.
For more details see that issue, where I add additional instructions.
